android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboardDownloaderActivity.java

```diff
@@ -384,6 +393,67 @@ public static boolean isCustom(String u) {
     return ret;
   }
 
+  // If a font JSONObject contains multiple font font files, only keep the .ttf source
+  private static void findTTF(JSONObject jsonFont) {
+    boolean updateJsonFont = false;
+    try {
+      JSONArray fontSource = jsonFont.optJSONArray(KMManager.KMKey_FontSource);
+      if (fontSource != null) {
+        boolean containsTTF = hasTTFFont(fontSource);
+        if (containsTTF) {
+          for (int i = fontSource.length() - 1; i >= 0; i--) {
+            String s = fontSource.getString(i);
+            if (!FileUtils.isTTFFont(s)) {
+              updateJsonFont = true;
+              if (Build.VERSION.SDK_INT > 19) {
+                fontSource.remove(i);
+              } else {
+                fontSource = removeJsonObjectAtIndex(fontSource, i);
+              }
+            }
+          }
+
+          if (updateJsonFont) {
```

Comment:
Have you tested a scenario where `findTTF` removes all entries from the array (e.g. only includes a .woff and .mobileconfig or something like that)?

Reply:
This is inside the `if (containsTTF) {` block so in that scenario, nothing gets removed.